Could not locate mcp_clj/http__init.class, mcp_clj/http.clj or mcp_clj/http.cljc

Hi Hugo,


Thanks for sharing your work on implementing MCP in Clojure! I'm trying to start an SSE MCP server and run into the following exception:


`java.io.FileNotFoundException: Could not locate mcp_clj/http__init.class, mcp_clj/http.clj or mcp_clj/http.cljc on classpath.`

<details>

<summary>Stacktrace</summary>


```
clojure.lang.ExceptionInfo: Error in component :mcp-server in system com.stuartsierra.component.SystemMap calling #'com.stuartsierra.component/start
        at com.stuartsierra.component$try_action.invokeStatic(component.cljc:120)
        at com.stuartsierra.component$try_action.invoke(component.cljc:117)
        at com.stuartsierra.component$update_system$fn__15697.invoke(component.cljc:140)
        at clojure.lang.ArraySeq.reduce(ArraySeq.java:119)
        at clojure.core$reduce.invokeStatic(core.clj:6964)
        at clojure.core$reduce.invoke(core.clj:6947)
        at com.stuartsierra.component$update_system.invokeStatic(component.cljc:136)
        at com.stuartsierra.component$update_system.doInvoke(component.cljc:130)
        at clojure.lang.RestFn.invoke(RestFn.java:448)
        at com.stuartsierra.component$start_system.invokeStatic(component.cljc:165)
        at com.stuartsierra.component$start_system.invoke(component.cljc:156)
        at com.stuartsierra.component$start_system.invokeStatic(component.cljc:163)
        at com.stuartsierra.component$start_system.invoke(component.cljc:156)
        at com.stuartsierra.component.SystemMap.start(component.cljc:181)
        at com.stuartsierra.component$eval15626$fn__15642$G__15614__15644.invoke(component.cljc:5)
        at com.stuartsierra.component$eval15626$fn__15642$G__15613__15647.invoke(component.cljc:5)
        at clojure.lang.AFn.applyToHelper(AFn.java:154)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.Var.alterRoot(Var.java:310)
        at clojure.core$alter_var_root.invokeStatic(core.clj:5563)
        at clojure.core$alter_var_root.doInvoke(core.clj:5558)
        at clojure.lang.RestFn.invoke(RestFn.java:428)
        at com.stuartsierra.component.repl$start.invokeStatic(repl.clj:38)
        at com.stuartsierra.component.repl$start.invoke(repl.clj:34)
        at tools.dev$after_refresh.invokeStatic(dev.clj:20)
        at tools.dev$after_refresh.invoke(dev.clj:16)
        at cider.nrepl.middleware.util.reload$resolve_and_invoke.invokeStatic(reload.clj:43)
        at cider.nrepl.middleware.util.reload$resolve_and_invoke.invoke(reload.clj:26)
        at cider.nrepl.middleware.util.reload$after_reply.invokeStatic(reload.clj:64)
        at cider.nrepl.middleware.util.reload$after_reply.invoke(reload.clj:57)
        at cider.nrepl.middleware.refresh$after_reply.invokeStatic(refresh.clj:72)
        at cider.nrepl.middleware.refresh$after_reply.invoke(refresh.clj:69)
        at cider.nrepl.middleware.refresh$refresh_reply$fn__15215$fn__15216.invoke(refresh.clj:98)
        at cider.nrepl.middleware.refresh$refresh_reply$fn__15215.invoke(refresh.clj:86)
        at nrepl.middleware.session$session_exec$session_loop__5448.invoke(session.clj:230)
        at nrepl.SessionThread.run(SessionThread.java:21)
Caused by: clojure.lang.ExceptionInfo: Transport factory failed
        at mcp_clj.server_transport.factory$create_transport.invokeStatic(factory.clj:80)
        at mcp_clj.server_transport.factory$create_transport.invoke(factory.clj:62)
        at mcp_clj.mcp_server.core$create_server.invokeStatic(core.clj:593)
        at mcp_clj.mcp_server.core$create_server.invoke(core.clj:522)
        at tools.mcp.Server.start(mcp.clj:20)
        at com.stuartsierra.component$eval15626$fn__15642$G__15614__15644.invoke(component.cljc:5)
        at com.stuartsierra.component$eval15626$fn__15642$G__15613__15647.invoke(component.cljc:5)
        at clojure.lang.AFn.applyToHelper(AFn.java:154)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.Var.applyTo(Var.java:707)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$apply.invoke(core.clj:662)
        at com.stuartsierra.component$try_action.invokeStatic(component.cljc:118)
        ... 35 common frames omitted
Caused by: clojure.lang.Compiler$CompilerException: Syntax error macroexpanding at (mcp_clj/json_rpc/sse_server.clj:1:1).
        at clojure.lang.Compiler.load(Compiler.java:8235)
        at clojure.lang.RT.loadResourceScript(RT.java:401)
        at clojure.lang.RT.loadResourceScript(RT.java:392)
        at clojure.lang.RT.load(RT.java:479)
        at clojure.lang.RT.load(RT.java:444)
        at clojure.core$load$fn__6933.invoke(core.clj:6189)
        at clojure.core$load.invokeStatic(core.clj:6188)
        at clojure.core$load.doInvoke(core.clj:6172)
        at clojure.lang.RestFn.invoke(RestFn.java:411)
        at clojure.core$load_one.invokeStatic(core.clj:5961)
        at clojure.core$load_one.invoke(core.clj:5956)
        at clojure.core$load_lib$fn__6875.invoke(core.clj:6003)
        at clojure.core$load_lib.invokeStatic(core.clj:6002)
        at clojure.core$load_lib.doInvoke(core.clj:5981)
        at clojure.lang.RestFn.applyTo(RestFn.java:145)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$load_libs.invokeStatic(core.clj:6044)
        at clojure.core$load_libs.doInvoke(core.clj:6028)
        at clojure.lang.RestFn.applyTo(RestFn.java:140)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$require.invokeStatic(core.clj:6066)
        at clojure.core$require.doInvoke(core.clj:6066)
        at clojure.lang.RestFn.invoke(RestFn.java:411)
        at mcp_clj.server_transport.factory$create_sse_server.invokeStatic(factory.clj:110)
        at mcp_clj.server_transport.factory$create_sse_server.invoke(factory.clj:108)
        at mcp_clj.server_transport.factory$create_transport.invokeStatic(factory.clj:78)
        ... 47 common frames omitted
Caused by: java.io.FileNotFoundException: Could not locate mcp_clj/http__init.class, mcp_clj/http.clj or mcp_clj/http.cljc on classpath. Please check that namespaces with dashes use underscores in the Clojure file name.
        at clojure.lang.RT.load(RT.java:482)
        at clojure.lang.RT.load(RT.java:444)
        at clojure.core$load$fn__6933.invoke(core.clj:6189)
        at clojure.core$load.invokeStatic(core.clj:6188)
        at clojure.core$load.doInvoke(core.clj:6172)
        at clojure.lang.RestFn.invoke(RestFn.java:411)
        at clojure.core$load_one.invokeStatic(core.clj:5961)
        at clojure.core$load_one.invoke(core.clj:5956)
        at clojure.core$load_lib$fn__6875.invoke(core.clj:6003)                                                                    at clojure.core$load_lib.invokeStatic(core.clj:6002)
        at clojure.core$load_lib.doInvoke(core.clj:5981)
        at clojure.lang.RestFn.applyTo(RestFn.java:145)
        at clojure.core$apply.invokeStatic(core.clj:669)                                                                           at clojure.core$load_libs.invokeStatic(core.clj:6044)
        at clojure.core$load_libs.doInvoke(core.clj:6028)
        at clojure.lang.RestFn.applyTo(RestFn.java:140)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$require.invokeStatic(core.clj:6066)
        at clojure.core$require.doInvoke(core.clj:6066)
        at clojure.lang.RestFn.invoke(RestFn.java:622)
        at mcp_clj.json_rpc.sse_server$eval17917$loading__6814__auto____17918.invoke(sse_server.clj:1)
        at mcp_clj.json_rpc.sse_server$eval17917.invokeStatic(sse_server.clj:1)
        at mcp_clj.json_rpc.sse_server$eval17917.invoke(sse_server.clj:1)
        at clojure.lang.Compiler.eval(Compiler.java:7757)
        at clojure.lang.Compiler.eval(Compiler.java:7746)
        at clojure.lang.Compiler.load(Compiler.java:8223)
        ... 72 common frames omitted
```
</details>

This is with a very minimal custom server:

``` clj
(def uuid-tool
  {:name           "uuid"
   :description    "Generate a random UUID"
   :inputSchema    {:type       "object"
                    :properties {}}
   :implementation (fn [_]
                     {:content [{:type "text" :text (str (random-uuid))}]
                      :isError false})})

(defrecord Server [port server]
  component/Lifecycle
  (start [this]
    (assoc this :server (server/create-server {:transport   {:port port
                                                             :type :sse}
                                               :server-info {:name    "name"
                                                             :title   "title"
                                                             :version "1.0.0"}
                                               :tools       {"uuid" uuid-tool}})))
  (stop [this]
    (when-let [stop-fn (:stop server)]
      (stop-fn))
    (assoc this :server nil)))
```

``` clj
org.hugoduncan/mcp-clj
{:deps/root "projects/server"
 :git/sha   "08ec63baeed813bc09673e61c20df8884961c11a"
 :git/url   "https://github.com/hugoduncan/mcp-clj"}
```

It looks like the `clj-mcp.http` component isn't included in any of the associated bases or components when pulling down a `projects/server` configuration.

--- a/components/http-server/deps.edn
+++ b/components/http-server/deps.edn
@@ -1,4 +1,5 @@
-{:deps {poly/http {:local/root "../http"}}
+{:deps {poly/http {:local/root "../http"}
+        poly/log {:local/root "../log"}}
  :aliases
  {:test {:extra-paths ["test"]
          :extra-deps  {hato/hato {:mvn/version "1.0.0"}}}}}

--- a/components/json-rpc/deps.edn
+++ b/components/json-rpc/deps.edn
@@ -1,7 +1,10 @@
 {:paths ["src"]
  :deps {org.clojure/clojure {:mvn/version "1.12.0"}
+        poly/http {:local/root "../http"}
         poly/http-client {:local/root "../http-client"}
-        poly/json {:local/root "../json"}}
+        poly/http-server {:local/root "../http-server"}
+        poly/json {:local/root "../json"}
+        poly/log {:local/root "../log"}}
  :aliases
  {:dev {:extra-paths ["test"]
         :extra-deps {hato/hato {:mvn/version "1.0.0"}}}

--- a/components/json-rpc/test/mcp_clj/json_rpc/deps_test.clj
+++ b/components/json-rpc/test/mcp_clj/json_rpc/deps_test.clj
@@ -1,0 +1,18 @@
+(ns mcp-clj.json-rpc.deps-test
+  "Test to verify json-rpc component dependencies are correctly declared"
+  (:require
+    [clojure.test :refer [deftest is testing]]))
+
+(deftest sse-server-dependencies-test
+  ;; Test that SSE server and its required dependencies can be loaded.
+  ;; This validates that components/json-rpc/deps.edn correctly declares
+  ;; all dependencies needed by the sse-server namespace.
+  (testing "SSE server namespace"
+    (testing "can be loaded with declared dependencies"
+      (is (nil? (require 'mcp-clj.json-rpc.sse-server))
+          "sse-server namespace should load successfully"))
+    (testing "required namespaces are available"
+      (is (nil? (require 'mcp-clj.http))
+          "mcp-clj.http should be available (required by sse-server:4)")
+      (is (nil? (require 'mcp-clj.http-server.adapter))
+          "mcp-clj.http-server.adapter should be available (required by sse-server:5)"))))

